infra/package-audit/src/engine.ts

```
@@ -0,0 +1,99 @@
+const zip = <T extends Record<string, unknown>>(obj: T): Partial<T> =>
+  Object.keys(obj).reduce((acc, k) => {
+    const v = obj[k];
+    if (
+      // Empty object
+      typeof v === 'undefined' ||
+      // Empty array
+      (Array.isArray(v) && v.length === 0)
+    ) {
+      return acc;
+    }
+    return { ...acc, [k]: v };
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)